server/expression/literal.go

```diff
@@ -53,7 +53,7 @@ func NewIntegerLiteral(integerValue string) (*expression.Literal, error) {
 
 // NewNullLiteral returns a new *expression.Literal containing a null value.
 func NewNullLiteral() *expression.Literal {
-	return expression.NewLiteral(nil, pgtypes.Unknown)
+	return expression.NewLiteral(nil, pgtypes.Null)
```

Comment:
This is not correct, and it's reintroducing a type that we removed almost a year ago:
* https://github.com/dolthub/doltgresql/pull/617

To reinforce this, the following query returns `unknown` as the type, which was the original value.
```sql
SELECT pg_typeof(NULL);
```

To drive it home even further, there's the official documentation: https://www.postgresql.org/docs/15/datatype-pseudo.html
> (unknown) Identifies a not-yet-resolved type, e.g., of an undecorated string literal.